### PR TITLE
Mostly javadocs, few changes (read description)

### PR DESCRIPTION
**Pull request by sbowe1**
- updated password requirements: removed number requirement + adjusted size limit to 8-64 char
- refactored User username -> name to match the front end
     - because full names are not unique, uniqueness requirement was removed from model (database) and as a check
- login DTO username + password -> email and password since we aren't using username anymore
- multiple users should be able to create plans of the same name -> when creating a plan, plan name check also takes the user ID into account
- fleshed out javadocs to include small description of all @params and @throws. also updated the descriptions to include more detail when necessary

### CODE REVIEW

File: p1-backend/src/main/java/com/example/p1_backend/repositories/PlanDao.java
Comment (pekellogg) [inline p1-backend/src/main/java/com/example/p1_backend/repositories/PlanDao.java:1]: Remove excessive useruserid unless required by JPA

Optional<Plan> getByNameAndUserUserId(String name, int userId); --> Optional<Plan> getByNameAndUserId(String name, int userId);

File: p1-backend/src/main/java/com/example/p1_backend/repositories/QuestionDao.java
Comment (pekellogg) [inline p1-backend/src/main/java/com/example/p1_backend/repositories/QuestionDao.java:1]: Remove excessive useruserid unless required by JPA

List<Question> findAllByUserUserId(int userId); --> List<Question> findAllByUserId(int userId);

File: p1-backend/src/main/java/com/example/p1_backend/repositories/ResourceDao.java
Comment (pekellogg) [inline p1-backend/src/main/java/com/example/p1_backend/repositories/ResourceDao.java:1]: Remove excessive topictopicid unless required by JPA


List<Resource> findAllByTopicTopicId(int topicId); --> List<Resource> findAllByTopicId(int topicId);

File: p1-backend/src/main/java/com/example/p1_backend/repositories/SubtopicDao.java
Comment (pekellogg) [inline p1-backend/src/main/java/com/example/p1_backend/repositories/SubtopicDao.java:1]: Remove excessive topictopicid unless required by JPA


List<Subtopic> findAllByTopicTopicId(int topicId); -->List<Subtopic> findAllByTopicId(int topicId);

File: p1-backend/src/main/java/com/example/p1_backend/repositories/TopicDao.java
Comment (pekellogg) [inline p1-backend/src/main/java/com/example/p1_backend/repositories/TopicDao.java:1]: List<Topic> findAllByPlanPlanId(int planId); --> List<Topic> findAllByPlanId(int planId);

**Review by pekellogg — CHANGES_REQUESTED** [review]
Requested changes related to method naming conventions; feel free to push back if required  format for JPA/Spring

File: p1-backend/src/main/java/com/example/p1_backend/repositories/PlanDao.java
Comment (sbowe1) [reply p1-backend/src/main/java/com/example/p1_backend/repositories/PlanDao.java:1]: required by JPA!

File: p1-backend/src/main/java/com/example/p1_backend/repositories/QuestionDao.java
Comment (pekellogg) [reply p1-backend/src/main/java/com/example/p1_backend/repositories/QuestionDao.java:1]: required by JPA!

File: p1-backend/src/main/java/com/example/p1_backend/repositories/ResourceDao.java
Comment (pekellogg) [reply p1-backend/src/main/java/com/example/p1_backend/repositories/ResourceDao.java:1]: required by JPA!

File: p1-backend/src/main/java/com/example/p1_backend/repositories/SubtopicDao.java
Comment (pekellogg) [reply p1-backend/src/main/java/com/example/p1_backend/repositories/SubtopicDao.java:1]: required by JPA!

File: p1-backend/src/main/java/com/example/p1_backend/repositories/TopicDao.java
Comment (pekellogg) [reply p1-backend/src/main/java/com/example/p1_backend/repositories/TopicDao.java:1]: required by JPA!

**Review by pekellogg — APPROVED** [review]

**Pull request merged**